cylc/flow/scripts/view.py

```diff
@@ -18,11 +18,16 @@
 
 """cylc view [OPTIONS] ARGS
 
-Print a processed workflow configuration.
+Print workflow configurations as processed before full parsing by Cylc. This
+includes Jinja2 or Empy template processing, and inlining of include-files.
+Some explanatory markup may also be requested.
+
+Warning: this command will fail if Jinja2 `CYLC_` variables are referenced
+without default values, because they are only defined during full parsing.
+E.g., `{{CYLC_WORKFLOW_NAME | default("not defined")}}`.
```

Comment:
Headings ending in `:` get put in bold which will help make this warning stand out from the surrounding text.

Swapped `CYLC_WORKFLOW_NAME` for `ID` as I can't really think of legit uses of `CYLC_WORKFLOW_NAME` so would rather keep the visibility of this low (we only added it to support MASS use cases at the MO).

```suggestion
Warning:
  This command will fail if Jinja2 `CYLC_` variables are referenced
  without default values, because they are only defined during full parsing.
  E.g., `{{CYLC_WORKFLOW_ID | default("not defined")}}`.
```